scripts/etc/init.d/rcS

```diff
@@ -118,6 +118,30 @@ mdev -s
 
 klogd -c 1
 
+# Check if there's an alternative rcS file and excute it
+# instead of this file. Only do this if this isn't the
+# alternative script already
+if [ -z ${am_subscript} ]; then
+    mkdir -p /boot
+    mount $bootpartition /boot
+    if [ -e /boot/config/installer/rcS ]; then
+        cp /boot/config/installer/rcS /rcS
+        umount /boot
+        echo "============================================="
+        echo "=== Start executing alternative rcS ========="
+        echo "---------------------------------------------"
+        export am_subscript=true
+        source /rcS
+        echo "---------------------------------------------"
+        echo "=== Execution of alternative rcS finished ==="
+        echo "============================================="
+        ${final_action} || reboot || exit
+    else
+        # Clean up, so the rest of the script continues as expected
+        umount /boot
+    fi
+fi
+
```

Comment:
I'd put this even earlier, if possible... the point is to "replace" the original rcS and be able to do development work on the alternative file (so not to build the image every time a change is made), so the less is done by the original rcS file, the better...

also, the logging would be overridden by the alternative rcS file anyway